lib/cog/nested_file.ex

```diff
@@ -0,0 +1,86 @@
+defmodule Cog.NestedFile do
+  @moduledoc """
+  Stores an arbitrary String on the filesystem of the Cog host in a file
+  named by combining the provided key and extension. Files are written to
+  a directory hierarchy that is created by splitting the filename into
+  two character segments and joining up to three of those segments with a
+  provided list of base path segments.
+
+  Note: The key is sanitized to remove dangerous characters, but the
+  base paths and extension are not. Make sure not to use user entered
+  values for these unsafe arguments.
```

Comment:
So is the `key` the only thing that's user-specified? The wording of the comment led me to believe otherwise.